### DISCUSSION TITLE
Working with Dates in Pocketbase hooks

### DISCUSSION THREAD

**Discussion by Vaxance** [pocketbase/pocketbase#4692] [Q&A] [answered]
Hello everyone,
I am trying to calculate a date from an entry in the DB in a hook. What exactly do I want?
I want to add x years to the date from the entry and then compare whether this calculated date is in the future.

Im using v0.22.2 and extend Pocketbase with JS.

Thanks for your help already.

Greetings,
Vaxance

**Comment by ganigeorgiev** [comment, marked as answer]
There are several ways to do it depending on your actual use case. You can use the Go translated [time.Time](https://pkg.go.dev/time#Time) or you can also create a regular JS `Date` object from the record value and interact with that but note that you'll have to replace the space separator with `T` to convert the string to ISO8601.

For example, to add 2 years to the current date you can try something like::

```js
onRecordBeforeCreateRequest((e) => {
    // variant 1 (using Go Time)
    let v1 = e.record.getTime("yourDateField")
    v1 = v1.addDate(2, 0, 0) // adds 2 years, 0 months and 0 days to the existing date
    e.record.set("yourDateField", v1)

    // variant 2 (using JS Date)
    let v2 = new Date(e.record.getString("yourDateField").replace(" ", "T"))
    v2.setFullYear(v2.getFullYear() + 2);
    e.record.set("yourDateField", v2)
}, "yourCollection")
```

**Reply by Vaxance** [reply]
Thanks for your fast answer, it was very helpfull.
